File: solutions/elixir/word-count/lib/word_count.ex
defmodule WordCount do
  @spec count(String.t()) :: map
  @doc """
  Count the number of words in the sentence.

  Words are compared case-insensitively.
  """

  def count(words) do
    words
    |> String.downcase()
    |> String.replace(~r/[.,:;!&@$%^]/, "")
    |> String.split([" ", "_"], trim: true)
    |> Enum.frequencies()
  end
end

File: solutions/julia/word-count/word-count.jl
function wordcount(sentence)
  words = String.(filter(!isempty, strip.(split(lowercase(sentence), r"[^a-z0-9']+"), '\'')))
  reduce(build_bag, words, init=Dict())
end

function build_bag(acc, n)
  haskey(acc, n) ? acc[n] += 1 : acc[n] = 1
  acc
end